### PR TITLE
Archive openj9-docs if ARCHIVE_JAVADOC is set

### PR DESCRIPTION
**Pull request by lumpfish**
Signed-off-by: Simon Rushton <rushton@uk.ibm.com>

### CODE REVIEW

**Comment by lumpfish** [comment]
@AdamBrousseau - is there some way for me to test this change to build.groovy?  It is to archive the openj9 only version of javadoc created by PR https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/307, which would then mean I can check it's created OK on all platforms.

**Comment by pshipton** [comment]
Hopefully this will do it. https://ci.eclipse.org/openj9/job/Pipeline-Release-Build/208/

**Comment by AdamBrousseau** [comment]
You'll have to use the internal personal build and set the SCM REPO and BRANCH parameters. Since the PR builds don't have ARCHIVE_JAVADOC set, this code won't get executed.l in a PR build. The build Pete linked will only use your repo for the OpenJ9 code, not the pipeline being checked out.

**Comment by pshipton** [comment]
Arg, trying again https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK11_s390x_linux/1783

**Comment by pshipton** [comment]
The build produced the desired artifact https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDK11_s390x_linux_Personal/610/OpenJ9-JDK11-Javadoc-openj9-s390x_linux-20200528-164427.tar.gz

**Comment by pshipton** [comment]
Trying 8 as well https://ci.eclipse.org/openj9/job/Pipeline-Release-Build/209/

**Comment by pshipton** [comment]
It didn't work for 8. https://ci.eclipse.org/openj9/job/Build_JDK8_s390x_linux_Personal/651/

**Comment by lumpfish** [comment]
Can't see why jdk11 would have worked but not jdk8.
The `Using JAVADOC_OPENJ9_ONLY_FILENAME = ` is unconditional, so if the code is being executed, that message should be seen.

I notice the pipelines were submitted differently.

jdk11 (worked): https://ci.eclipse.org/openj9/job/Pipeline-Release-Build/208/parameters/
```
VENDOR_REPO blank
VENDOR_BRANCH blank
```
jdk8 (did not work): https://ci.eclipse.org/openj9/job/Pipeline-Release-Build/209/parameters/
```
VENDOR_REPO https://github.com/lumpfish/openj9.git
VENDOR_BRANCH archive_openj9_only_javadoc
```

**Comment by lumpfish** [comment]
Regarding jdk11 vs. jdk8, the `Using JAVADOC_OPENJ9_ONLY_FILENAME =` is unconditional, so if the code is being executed, that message should be seen.

I notice the pipelines were submitted differently.

jdk11 (worked): https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK11_s390x_linux/1783/parameters/
```
SCM_BRANCH	archive_openj9_only_javadoc				
SCM_REPO	https://github.com/lumpfish/openj9.git				
VENDOR_BRANCH					
VENDOR_REPO					
```
jdk8 (did not work): https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK8_s390x_linux/1813/parameters/
```
SCM_BRANCH	refs/heads/master			
SCM_REPO	https://github.com/eclipse/openj9.git			
VENDOR_BRANCH	archive_openj9_only_javadoc			
VENDOR_REPO	https://github.com/lumpfish/openj9.git			
```

**Comment by pshipton** [comment]
Started a new jdk8 build with the SCM parameter set.
https://ci.eclipse.org/openj9/job/Build_JDK8_s390x_linux_Personal/657/

**Comment by pshipton** [comment]
Thanks Simon. The jdk8 build worked after setting the SCM parameters.

**Comment by pshipton** [comment]
@keithc-ca fyi

File: buildenv/jenkins/common/build.groovy [hunk 1]
```diff
@@ -370,13 +383,19 @@ def archive_sdk() {
                         currentBuild.description += "<br><a href=${JAVADOC_LIB_URL}>${JAVADOC_FILENAME}</a>"
                         echo "Javadoc:'${JAVADOC_LIB_URL}"
                     }
+                    if (fileExists("${JAVADOC_OPENJ9_ONLY_FILENAME}")) {
+                        JAVADOC_OPENJ9_ONLY_LIB_URL = "${ARTIFACTORY_CONFIG[ARTIFACTORY_CONFIG['defaultGeo']]['url']}/${ARTIFACTORY_CONFIG['uploadDir']}${JAVADOC_OPENJ9_ONLY_FILENAME}"
+                        currentBuild.description += "<br><a href=${JAVADOC_OPENJ9_ONLY_LIB_URL}>${JAVADOC_OPENJ9_ONLY_FILENAME}</a>"
+                        echo "Javadoc (OpenJ9 extensions only):'${JAVADOC_OPENJ9_ONLY_LIB_URL}"
```
Comment (keithc-ca) [inline buildenv/jenkins/common/build.groovy:389]: As I had another look at this today, I noticed that this message is missing the closing single quote just like line 384: could you fix both, please?

**Comment by keithc-ca** [comment]
@lumpfish Please squash in preparation for merging.

**Comment by lumpfish** [comment]
Squashed

**Pull request merged**